__tests__/server/plugins/addSecurityHeaders.spec.js

```diff
@@ -14,39 +14,100 @@
  * permissions and limitations under the License.
  */
 
-import addSecurityHeaders from '../../../src/server/middleware/addSecurityHeaders';
-
-describe('addSecurityHeaders', () => {
-  it('should add all expected security headers', () => {
-    const req = { get: jest.fn(), headers: {} };
-    const res = { set: jest.fn((key, value) => value) };
-    const next = jest.fn();
-    addSecurityHeaders(req, res, next);
-
-    const securityHeaders = {
-      'X-Frame-Options': 'DENY',
-      'X-Content-Type-Options': 'nosniff',
-      'Strict-Transport-Security': 'max-age=15552000; includeSubDomains',
-      'X-XSS-Protection': '1; mode=block',
-      'Referrer-Policy': 'same-origin',
-    };
-    expect(res.set.mock.calls.length).toEqual(Object.keys(securityHeaders).length);
-    Object.keys(securityHeaders).forEach(
-      (header) => expect(res.set).toBeCalledWith(header, securityHeaders[header])
-    );
-    expect(next).toBeCalled();
-  });
-
-  describe('Referrer-Policy', () => {
-    it('default can be overridden', () => {
-      const req = { get: jest.fn(), headers: {} };
-      const res = { set: jest.fn((key, value) => value) };
-      process.env.ONE_REFERRER_POLICY_OVERRIDE = 'no-referrer';
-
-      addSecurityHeaders(req, res, jest.fn());
-      expect(res.set).toBeCalledWith('Referrer-Policy', 'no-referrer');
-
-      delete process.env.ONE_REFERRER_POLICY_OVERRIDE;
-    });
-  });
+import addSecurityHeaders from '../../../src/server/plugins/addSecurityHeaders';
+
+beforeEach(() => {
+  delete process.env.ONE_REFERRER_POLICY_OVERRIDE;
+});
+
+test('adds security headers', () => {
+  const request = {
+    headers: {},
+  };
+  const reply = {
+    header: jest.fn(),
+  };
+  const fastify = {
+    addHook: jest.fn(async (_hook, cb) => {
+      await cb(request, reply);
+    }),
+  };
+  const done = jest.fn();
+
+  addSecurityHeaders(fastify, {}, done);
+
+  expect(fastify.addHook).toHaveBeenCalled();
+  expect(done).toHaveBeenCalled();
+  expect(reply.header).toHaveBeenCalledTimes(8);
```

Comment:
Could have used foreach like on existing? 
    `expect(res.set.mock.calls.length).toEqual(Object.keys(securityHeaders).length);
    Object.keys(securityHeaders).forEach(...`

Reply:
we could, what do you think is the advantage of the approach you're proposing?

Reply:
Just cleaner approach in my view. But this works.